test/parallel/test-buffer-readintbe.js

```diff
@@ -0,0 +1,24 @@
+'use strict';
+require('../common');
+const assert = require('assert');
+
+// testing basic functionality of readIntBE()
+
+const buf = Buffer.from([42, 84, 167, 127]);
+const result = buf.readIntBE(1);
+
+assert.strictEqual(result, 42);
+
+assert.throws(
+  () => {
+    buf.readIntBE(5);
+  },
+  /^RangeError: Index out of range$/
+);
+
+assert.doesNotThrow(
+  () => {
+    buf.readIntBE(5, 0, true);
```

Comment:
Isn't the point of this test to provide invalid args, and show that if `noAssert` is true, invalid args don't throw?

I am absolutely baffled as to how not asserting on an offset past the end of the buffer is considered a feature, even an optional one, and further, the docs are absolutely silent on what the return value is when noAssert is specified... is it an explicit way to read arbitrary memory locations past the end of buffer? Will `undefined` be returned? Can't tell from the docs.